src/main/java/com/verisign/storm/metrics/reporters/KafkaReporter.java

```diff
@@ -0,0 +1,152 @@
+ */
+package com.verisign.storm.metrics.reporters;
+
+import com.google.common.base.Throwables;
+import com.verisign.ie.styx.avro.graphingMetrics.GraphingMetrics;
+import com.verisign.storm.metrics.graphite.ConnectionFailureException;
+import com.verisign.storm.metrics.graphite.GraphiteCodec;
+import kafka.javaapi.producer.Producer;
+import kafka.producer.KeyedMessage;
+import kafka.producer.ProducerConfig;
+import org.apache.avro.io.BinaryEncoder;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.io.EncoderFactory;
+import org.apache.avro.specific.SpecificDatumWriter;
+import org.apache.avro.specific.SpecificRecordBase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * This class encapsulates an Apache Kafka producer, sending generated metrics into a configurable Kafka topic.
+ */
+public class KafkaReporter extends AbstractReporter {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaReporter.class);
+  public static final String KAFKA_TOPIC_NAME_FIELD = "metrics.kafka.topic";
+  public static final String KAFKA_BROKER_LIST_FIELD = "metrics.kafka.metadata.broker.list";
+  public static final String KAFKA_METADATA_BROKER_LIST = "metadata.broker.list";
+
+  private String kafkaTopicName;
+  private String kafkaBrokerList;
+  private LinkedList<GraphingMetrics> buffer;
+
+  private Producer<String, byte[]> kafkaProducer;
+
+  private int failures;
+
+  public KafkaReporter(Map conf) {
+    super(conf);
+
+    if (conf.containsKey(KAFKA_TOPIC_NAME_FIELD)) {
+      kafkaTopicName = (String) conf.get(KAFKA_TOPIC_NAME_FIELD);
+    }
+    else {
+      throw new IllegalArgumentException("Field " + KAFKA_TOPIC_NAME_FIELD + " required.");
+    }
+
+    if (conf.containsKey(KAFKA_BROKER_LIST_FIELD)) {
+      kafkaBrokerList = (String) conf.get(KAFKA_BROKER_LIST_FIELD);
+      conf.put(KAFKA_METADATA_BROKER_LIST, kafkaBrokerList);
+    }
+    else if (conf.containsKey(KAFKA_METADATA_BROKER_LIST)) {
+      kafkaBrokerList = (String) conf.get(KAFKA_METADATA_BROKER_LIST);
+    }
+    else {
+      throw new IllegalArgumentException("Field " + KAFKA_BROKER_LIST_FIELD + " required.");
+    }
+
+    Properties producerProps = new Properties();
+    for (String key : ((Map<String, Object>) conf).keySet()) {
+      producerProps.setProperty(key, (String) conf.get(key));
+    }
+
+    kafkaProducer = new Producer<String, byte[]>(new ProducerConfig(producerProps));
+    buffer = new LinkedList<GraphingMetrics>();
+    failures = 0;
+  }
+
+  @Override public void connect() throws ConnectionFailureException {
+  }
+
+  @Override public void disconnect() throws ConnectionFailureException {
+  }
+
+  @Override public void appendToBuffer(String prefix, Map<String, Object> metrics, long timestamp) {
```

Comment:
Should we change `Map<String, Object> metrics` to `Map<String, Double> metrics` to be more type safe?  The Avro schema for the Kafka backend, for example, only allows double anyways.